addons/mrp/models/stock_rule.py

```diff
@@ -198,7 +198,8 @@ def _get_lead_days(self, product, **values):
         if not manufacture_rule:
             return delay, delay_description
         manufacture_rule.ensure_one()
-        bom = values.get('bom') or (product.variant_bom_ids or product.bom_ids)[:1]
+        bom_ids = self.env['mrp.bom']._bom_find(product, picking_type=manufacture_rule.picking_type_id, company_id=manufacture_rule.company_id.id)[product]
+        bom = values.get('bom') or bom_ids
```

Comment:
```suggestion
        bom = values.get('bom') or self.env['mrp.bom']._bom_find(product, picking_type=manufacture_rule.picking_type_id, company_id=manufacture_rule.company_id.id)[product]
```
Please do the small opti before merging in master